resources/migration-guide-v2-v5.md

```diff
@@ -0,0 +1,444 @@
+
+### Key Differences
+
+1. **Module Structure**:
+   - The old `DeployUtil` module had multiple static utility methods.
+   - The new `Deploy` class organizes these functionalities as methods on a single object.
+2. **Serialization and Deserialization**:
+   - The new module uses `TypedJSON` for serialization/deserialization, ensuring type safety.
+3. **Improved Method Organization**:
+   - Methods are grouped logically, providing a cleaner and more intuitive API.
+
+#### Migration Examples
+
+##### Old Syntax: Creating a Deploy (2.x)
+
+```typescript
+import { DeployUtil } from 'casper-js-sdk';
+
+const deployParams = new DeployUtil.DeployParams(
+  senderKey.publicKey,
+  'casper-test'
+);
+const session = DeployUtil.ExecutableDeployItem.newTransfer(
+  10,
+  recipientKey.publicKey,
+  undefined,
+  1
+);
+const payment = DeployUtil.standardPayment(10000000000000);
+let deploy = DeployUtil.makeDeploy(deployParams, session, payment);
+deploy = DeployUtil.signDeploy(deploy, senderKey);
+
+return deploy;
+```
+
+##### New Syntax: Creating a Deploy (5.x)
+
+```typescript
+import {
+  Deploy,
+  DeployHeader,
+  ExecutableDeployItem
+} from 'casper-js-sdk';
+
+const session = new ExecutableDeployItem();
+
+session.transfer = TransferDeployItem.newTransfer(
+  10,
+  recipientKey.publicKey,
+  undefined,
+  1
+);
+
+const deployHeader = DeployHeader.default();
+deployHeader.account = senderKey.publicKey;
+deployHeader.chainName = 'casper-test';
+
+const payment = ExecutableDeployItem.standardPayment(10000000000000);
+const deploy = Deploy.makeDeploy(deployHeader, payment, session);
+deploy.sign(senderKey);
+```
+
+##### Old Syntax: Serialization and Deserialization (2.x)
+
+```typescript
+import { DeployUtil } from 'casper-js-sdk';
+
+const deployJson = DeployUtil.deployToJson(deploy);
+const deployFromJson = DeployUtil.deployFromJson(deployJson);
+```
+
+##### New Syntax: Serialization and Deserialization (5.x)
+
+```typescript
+import { Deploy } from 'casper-js-sdk';
+
+const deployJson = Deploy.toJSON(deploy);
+const deployFromJson = Deploy.fromJSON(deployJson);
+```
+
+### Additional Notes
+
+- The new `Deploy` class provides a more type-safe and modular approach, reducing errors and improving code readability.
+- Refer to the [Deploy Class Documentation](../src/types/Deploy.ts) for a complete list of available methods.
+- Ensure that you update your dependencies and verify compatibility with the 5.x SDK before migrating.
+
+---
+
+### 7. **`Contracts` Abstraction Removed**
+
+With the release of version 5.0 of the `casper-js-sdk`, we removed the old abstraction for creating and interacting with deploys, including the functions `install` and `callEntrypoint`. This guide provides detailed steps for migrating from the old abstraction to the new approach using the SDK's updated API.
+
```

Comment:
Should we mention here that for most common contracts (ie. auction, cep-18, cep78) there are specific deploy builders and link to `Base usage` in README.md where there're few examples?